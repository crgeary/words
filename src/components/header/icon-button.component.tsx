import { ComponentPropsWithoutRef, FC } from "react";

export type IconButtonProps = ComponentPropsWithoutRef<"button"> & {
    //
};

export const IconButton: FC<IconButtonProps> = ({ children }) => {
    return (
        <button className="p-2 rounded-full text-violet-800 hover:bg-violet-100 hover:text-violet-900">
            {children}
        </button>
    );
};
