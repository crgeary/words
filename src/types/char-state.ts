export enum CharState {
    Correct = "correct",
    Present = "present",
    Absent = "absent",
    Idle = "idle",
}
