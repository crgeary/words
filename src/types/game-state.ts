export enum GameState {
    Idle = "idle",
    Playing = "playing",
    Fail = "fail",
    Success = "success",
}
