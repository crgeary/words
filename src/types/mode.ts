export enum Mode {
    Random = "random",
    Wordle = "wordle",
}

export type ModeConfig = {
    [k in Mode]: { name: string };
};
