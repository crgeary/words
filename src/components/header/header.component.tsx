import { FC } from "react";
import { AdjustmentsIcon, MenuIcon, PresentationChartBarIcon } from "@heroicons/react/outline";
import { IconButton } from "./icon-button.component";
import { Mode } from "../../types/mode";
import { MODE_CONFIG } from "../../constants";

type HeaderProps = {
    mode?: Mode;
};

export const Header: FC<HeaderProps> = ({ mode }) => {
    return (
        <header
            role="banner"
            className="h-14 px-3 flex items-center justify-between bg-white border-t-2 border-b border-t-violet-800 border-b-violet-200"
        >
            <div className="w-20">
                <IconButton>
                    <MenuIcon className="h-5 w-5" />
                    <span className="sr-only">Menu</span>
                </IconButton>
            </div>
            <p className="text-center">
                <strong className="block text-lg font-semibold text-slate-700 leading-none">
                    Words
                </strong>
                {mode && (
                    <span className="block mt-1 text-xs text-slate-400 leading-none">
                        Mode: {MODE_CONFIG[mode].name}
                    </span>
                )}
            </p>
            <div className="w-20 flex justify-end">
                <IconButton>
                    <PresentationChartBarIcon className="h-5 w-5" />
                    <span className="sr-only">Stats</span>
                </IconButton>
                <IconButton>
                    <AdjustmentsIcon className="h-5 w-5" />
                    <span className="sr-only">Settings</span>
                </IconButton>
            </div>
        </header>
    );
};
