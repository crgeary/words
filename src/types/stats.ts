import { Mode } from "./mode";

export type StatsEntry = {
    mode: Mode;
    solution: string;
    guesses: string[];
    date: Date;
};
