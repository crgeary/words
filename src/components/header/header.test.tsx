import { render } from "@testing-library/react";

import { Mode } from "../../types/mode";
import { Header } from "./header.component";

test("it should match snapshot", () => {
    const { container } = render(<Header mode={Mode.Random} />);
    expect(container.firstChild).toMatchSnapshot();
});
