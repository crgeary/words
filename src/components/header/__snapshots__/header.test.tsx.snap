// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should match snapshot 1`] = `
<header
  class="h-14 px-3 flex items-center justify-between bg-white border-t-2 border-b border-t-violet-800 border-b-violet-200"
  role="banner"
>
  <div
    class="w-20"
  >
    <button
      class="p-2 rounded-full text-violet-800 hover:bg-violet-100 hover:text-violet-900"
    >
      <svg
        aria-hidden="true"
        class="h-5 w-5"
        fill="none"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 6h16M4 12h16M4 18h16"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
      <span
        class="sr-only"
      >
        Menu
      </span>
    </button>
  </div>
  <p
    class="text-center"
  >
    <strong
      class="block text-lg font-semibold text-slate-700 leading-none"
    >
      Words
    </strong>
    <span
      class="block mt-1 text-xs text-slate-400 leading-none"
    >
      Mode: 
      Random
    </span>
  </p>
  <div
    class="w-20 flex justify-end"
  >
    <button
      class="p-2 rounded-full text-violet-800 hover:bg-violet-100 hover:text-violet-900"
    >
      <svg
        aria-hidden="true"
        class="h-5 w-5"
        fill="none"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 13v-1m4 1v-3m4 3V8M8 21l4-4 4 4M3 4h18M4 4h16v12a1 1 0 01-1 1H5a1 1 0 01-1-1V4z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
      <span
        class="sr-only"
      >
        Stats
      </span>
    </button>
    <button
      class="p-2 rounded-full text-violet-800 hover:bg-violet-100 hover:text-violet-900"
    >
      <svg
        aria-hidden="true"
        class="h-5 w-5"
        fill="none"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 6V4m0 2a2 2 0 100 4m0-4a2 2 0 110 4m-6 8a2 2 0 100-4m0 4a2 2 0 110-4m0 4v2m0-6V4m6 6v10m6-2a2 2 0 100-4m0 4a2 2 0 110-4m0 4v2m0-6V4"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
      <span
        class="sr-only"
      >
        Settings
      </span>
    </button>
  </div>
</header>
`;
